src/model/formula/Formula.ts

```diff
@@ -64,22 +64,27 @@ abstract class Formula extends Expression {
 
     for (const { sign, formula } of formulas) {
       let current = { sign: sign, formula: formula };
-      if (formula.eval(structure, e) !== sign) {
-        if (!shortest) {
-          shortest = current;
-          winning.push(shortest);
-        }
-
-        if (
-          shortest.formula.gameDepth(shortest.sign) > formula.gameDepth(sign)
-        ) {
-          shortest = current;
-          winning = [shortest];
-        } else if (
-          shortest.formula.gameDepth(shortest.sign) === formula.gameDepth(sign)
-        ) {
-          winning.push(current);
+      try {
+        if (formula.eval(structure, e) !== sign) {
+          if (!shortest) {
+            shortest = current;
+            winning.push(shortest);
+          }
+
+          if (
+            shortest.formula.gameDepth(shortest.sign) > formula.gameDepth(sign)
+          ) {
+            shortest = current;
+            winning = [shortest];
+          } else if (
+            shortest.formula.gameDepth(shortest.sign) ===
+            formula.gameDepth(sign)
+          ) {
+            winning.push(current);
+          }
         }
+      } catch (error) {
+        return winning;
```

Comment:
Nezda sa mi, ze toto je dobre miesto na odchytavanie vyhodnocovacich vynimiek. Ak sa podformula neda vyhodnotit (formula sa mohla dat, lebo na to nemusi byt nutne vyhodnocovat vsetky podformuly), problem je zrejme strukture alebo ohodnoteni – cize na strane pouzivatela. O tom sa musi dozvediet a vyriesit to.

Namiesto toho sa tu ticho vrati neuplne `winning`, ktore moze obmedzit volby Sveta a zaviest ho na cestu, na ktorej prehra. Pouzivatel si bude mylne mysliet, ze vsetko je v poriadku, pritom by mozno prehral, keby boli struktura a ohodnotenie uplne.